### PR TITLE
Clarify when the `changed` signal is emitted for `Resource` and `Material`

### LINKED ISSUE #69571: Materials do not fire `changed` signal when any of its properties change
### Godot version

4.0.beta7

### System information

Irelevant

### Issue description

Any Material ~~Node~~ Resource does not emit the `changed` signal when its properties change (Properties of base Resources still emit the signal)

### Steps to reproduce

Copy-paste this script to any node
```gdscript
@tool
extends Node


@export var x: Resource:
	set(value):
		print("set")
		x = value
		x.changed.connect(foo)


func foo():
	print("changed")
```

Try different Resources and change their properties and watch when the `changed` signal is emmited.
When testing any Material derivate, any of its properties changing did not emitted the signal.

### Minimal reproduction project

See above

### PR DESCRIPTION
**Pull request by josefalanga**
Proposed fix to [#69571](https://github.com/godotengine/godot/issues/69571)

* *Bugsquad edit, fixes: #69571*

### CODE REVIEW

File: doc/classes/Material.xml [hunk 1]
```diff
@@ -5,6 +5,9 @@
 	</brief_description>
 	<description>
 		Material is a base [Resource] used for coloring and shading geometry. All materials inherit from it and almost all [VisualInstance3D] derived nodes carry a Material. A few flags and parameters are shared between all material types and are configured here.
+		[b]Note:[/b]  For performance reasons the [code]changed[/code] signal, it's only emitted when these properties are changed:
```
Comment (YeldhamDev) [inline doc/classes/Material.xml]:
```suggestion
		[b]Note:[/b]  For performance reasons the [signal Resource.changed] signal, it's only emitted when these properties are changed:
```

Comment (AThousandShips) [inline doc/classes/Material.xml:8]:
```suggestion
		[b]Note:[/b]  For performance reasons the [signal Resource.changed] signal, it's only emitted when these properties are changed:
```

Comment (YeldhamDev) [reply doc/classes/Material.xml]: <s>I think my review above is enough. :stuck_out_tongue: </s>

I'm a imbecile who didn't commit his review. :upside_down_face:

Comment (josefalanga) [reply doc/classes/Material.xml]: done!

**Review by clayjohn — APPROVED** [review]

File: doc/classes/Material.xml [hunk 2]
```diff
@@ -5,6 +5,9 @@
 	</brief_description>
 	<description>
 		Material is a base [Resource] used for coloring and shading geometry. All materials inherit from it and almost all [VisualInstance3D] derived nodes carry a Material. A few flags and parameters are shared between all material types and are configured here.
+		[b]Note:[/b]  For performance reasons the [signal Resource.changed] signal, it's only emitted when these properties are changed:
+		- Name (inherited from [Resource])
+		- Shader
```
Comment (YuriSizov) [inline doc/classes/Material.xml:10]:
```suggestion
		[b]Note:[/b] For performance reasons the [signal Resource.changed] signal is only emitted when the name and the shader of the material are changed.
```

Although these should ideally link to specific properties or methods.

**Comment by YuriSizov** [comment]
Could you also please squash your commits into one? Make sure that the final commit has a short but descriptive message (the title of this PR is a good option). See [this documentation](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase), if you need help with squashing.

File: doc/classes/Material.xml (hunk 2 above)
Comment (josefalanga) [reply doc/classes/Material.xml:10]: I can't find usages of something like properties linking, like `[property Resource.resource_name]` in the rest of the docs, does that works? If not, what would be the best approach to this linking?

Also, the material change is only done for ShaderMaterials, and does nothing in non-editor builds, so maybe this clarification in Material is actually pointless.

Comment (YuriSizov) [reply doc/classes/Material.xml:10]: It's `[member property_name]` not `[property property_name]`.

Comment (YuriSizov) [reply doc/classes/Material.xml]: Yes.

**Comment by YuriSizov** [comment]
Thanks! And congrats on your first merged Godot pull request!

**Pull request merged**